pandas/core/strings/accessor.py

```diff
@@ -1394,12 +1394,13 @@ def fullmatch(self, pat, case: bool = True, flags: int = 0, na=None):
     @forbid_nonstring_types(["bytes"])
     def replace(
         self,
-        pat: str | re.Pattern,
-        repl: str | Callable,
+        pat: str | re.Pattern | None = None,
+        repl: str | Callable | None = None,
         n: int = -1,
         case: bool | None = None,
         flags: int = 0,
         regex: bool = False,
+        repl_kwargs: dict | None = None,
```

Comment:
I think instead of adding a new keyword, we want to allow `pat` to be a dictionary (in which case `repl` must be `None`).